docs/7-deployment/2-caching.md

```diff
@@ -1,4 +1,350 @@
+CACHE_L1_ENABLED=true
+CACHE_L1_MAX_SIZE=512
+
+# L2 Cache (shared Redis)
+CACHE_L2_ENABLED=true
+CACHE_L2_BACKEND=redis
+CACHE_L2_ADDRESSES=redis:6379
+CACHE_L2_PASSWORD=your-redis-password
+
+# Cache TTL
+CACHE_TTL=10m
+```
+
+### Using Memcached
+
+If you prefer Memcached over Redis:
+
+```bash
+# L1 Cache
+CACHE_L1_ENABLED=true
+CACHE_L1_MAX_SIZE=512
+
+# L2 Cache (Memcached)
+CACHE_L2_ENABLED=true
+CACHE_L2_BACKEND=memcached
+CACHE_L2_ADDRESSES=memcached:11211
+
+# Cache TTL
+CACHE_TTL=5m
+```
+
+### High Availability Redis Cluster
+
+For production deployments with Redis cluster:
+
+```bash
+# L1 Cache
+CACHE_L1_ENABLED=true
+CACHE_L1_MAX_SIZE=1024
+
+# L2 Cache (Redis Cluster)
+CACHE_L2_ENABLED=true
+CACHE_L2_BACKEND=redis
+CACHE_L2_ADDRESSES=redis-node-1:6379,redis-node-2:6379,redis-node-3:6379
+CACHE_L2_USERNAME=hugr
+CACHE_L2_PASSWORD=your-redis-password
+CACHE_L2_DATABASE=0
+
+# Cache TTL
+CACHE_TTL=15m
+```
+
+## Docker Compose Example with Redis
+
+Here's a complete example using Docker Compose with Redis cache:
+
+```yaml
+version: '3.8'
+
+services:
+  hugr:
+    image: ghcr.io/hugr-lab/server:latest
+    ports:
+      - "15000:15000"
+    environment:
+      - BIND=:15000
+      - ADMIN_UI=true
+      - CORE_DB_PATH=/data/core.db
+
+      # L1 Cache
+      - CACHE_L1_ENABLED=true
+      - CACHE_L1_MAX_SIZE=512
+
+      # L2 Cache (Redis)
+      - CACHE_L2_ENABLED=true
+      - CACHE_L2_BACKEND=redis
+      - CACHE_L2_ADDRESSES=redis:6379
+      - CACHE_L2_PASSWORD=redis_password
+
+      # Cache TTL
+      - CACHE_TTL=10m
+    volumes:
+      - ./data:/data
+    depends_on:
+      - redis
+
+  redis:
+    image: redis:7-alpine
+    command: redis-server --requirepass redis_password
+    ports:
+      - "6379:6379"
+    volumes:
+      - redis_data:/data
+
+volumes:
+  redis_data:
+```
+
+## Docker Compose Example with Memcached
+
+Alternative example using Memcached:
+
+```yaml
+version: '3.8'
+
+services:
+  hugr:
+    image: ghcr.io/hugr-lab/server:latest
+    ports:
+      - "15000:15000"
+    environment:
+      - BIND=:15000
+      - ADMIN_UI=true
+      - CORE_DB_PATH=/data/core.db
+
+      # L1 Cache
+      - CACHE_L1_ENABLED=true
+      - CACHE_L1_MAX_SIZE=512
+
+      # L2 Cache (Memcached)
+      - CACHE_L2_ENABLED=true
+      - CACHE_L2_BACKEND=memcached
+      - CACHE_L2_ADDRESSES=memcached:11211
+
+      # Cache TTL
+      - CACHE_TTL=10m
+    volumes:
+      - ./data:/data
+    depends_on:
+      - memcached
+
+  memcached:
+    image: memcached:1.6-alpine
+    ports:
+      - "11211:11211"
+    command: memcached -m 256
+```
+
+## Cache Behavior
+
+### Cache Key Generation
+
+Hugr automatically generates cache keys based on:
+- Query structure
+- Query parameters
+- User context (when authentication is enabled)
+- Data source configuration
+
+### Cache Invalidation
+
+Cache entries are invalidated when:
+- TTL expires
+- Data source schema is reloaded
+- Manual cache clear is triggered (via GraphQL mutation)
```

Comment:
or directive @invalidate_cache, look at the directive references doc file to provide cache directives explanations in this section